tests/models/bloom/test_modeling_bloom.py

```diff
@@ -377,20 +377,46 @@ def test_model_from_pretrained(self):
+        # We set allow_fp16_reduced_precision_reduction = True. Please see: https://pytorch.org/docs/stable/notes/cuda.html#reduced-precision-reduction-in-fp16-gemms
+        # This discrepancy is observed only when using small models and seems to be stable for larger models.
+        # Our conclusion is that these operations are flaky for small inputs but seems to be stable for larger inputs (for the functions `baddmm` and `bmm`), and therefore for larger models.
+
+        # Here is a summary of an ablation study of our observations
+        # EXPECTED_OUTPUT = "I enjoy walking with my cute dog, and I love to watch the kids play. I am a very active person, and I am a very good listener. I am a very good person, and I am a very good person. I am a"
+        # 350m + allow_fp16_reduced_precision_reduction = False  + torch.bmm  ==> PASS
+        # 350m + allow_fp16_reduced_precision_reduction = False  + torch.baddm  ==> PASS
+        # 350m + allow_fp16_reduced_precision_reduction = True  + torch.baddm  ==> PASS
+        # 350m + allow_fp16_reduced_precision_reduction = True  + torch.bmm  ==> FAIL
+
+        # EXPECTED_OUTPUT = "I enjoy walking with my cute dog, but I also enjoy hiking, biking, and swimming. I love to cook and bake. I love to cook and bake. I love to cook and bake. I love to cook and bake. I love"
+        # >=760m + allow_fp16_reduced_precision_reduction = True  + torch.baddm  ==> PASS  (for use_cache=True and use_cache=False)
+        # >=760m + allow_fp16_reduced_precision_reduction = True  + torch.bmm  ==> PASS
+        # >=760m + allow_fp16_reduced_precision_reduction = False  + torch.bmm  ==> PASS
+
         path_350m = "bigscience/bloom-350m"
         model = BloomForCausalLM.from_pretrained(path_350m, torch_dtype="auto", use_cache=True).cuda()
         model = model.eval()
         tokenizer = BloomTokenizerFast.from_pretrained(path_350m)
 
         input_sentence = "I enjoy walking with my cute dog"
-        EXPECTED_OUTPUT = (
+        EXPECTED_OUTPUT = (  # for bloom-350m
             "I enjoy walking with my cute dog, and I love to watch the kids play. I am a very active person, and I am"
             " a very good listener. I am a very good person, and I am a very good person. I am a"
         )
 
+        # EXPECTED_OUTPUT = (
+        #     "I enjoy walking with my cute dog, but I also enjoy hiking, biking, and swimming. I love to cook and bake."
+        #     " I love to cook and bake. I love to cook and bake. I love to cook and bake. I love"
+        # ) # for bloom-760m
+
         input_ids = tokenizer.encode(input_sentence, return_tensors="pt")
+        torch.backends.cuda.matmul.allow_fp16_reduced_precision_reduction = False
         greedy_output = model.generate(input_ids.cuda(), max_length=50)
+        torch.backends.cuda.matmul.allow_fp16_reduced_precision_reduction = True
 
+        print(tokenizer.decode(greedy_output[0], skip_special_tokens=True))
```

Comment:
Remove this line

Reply:
Should be fixed in ed9682dcdd7366e2a95a3e021c37a2434fda6a24